### PR TITLE
Add note on how to upgrade managed nodegroups using custom AMI

### LINKED ISSUE #5006: Managed node group upgrade using AWS customer AMI
Attempts to implementing a continuos process for the creation and update of eks cluster api and managed node groups using eksctl are not possible for nodegroups using customer/user AMIs.

This process uses our own AMIs that are produced from AWS's default amazon2 images that are patched with minimal addons.

The workflow for using eksctl does not allow for upgrading managed node groups (e.g 1.20) that have been created for an older controller api version (e.g 1.20) when the controller has been updated to 1.21.

Updating the ami used in the eksctl config.yml file does not upgrade the managed nodegroup even though updating the cluster version in the same config.yml file updates the controller api version.

We are therefore going to be forced to implement the creation of new managed node groups when upgrades to controller api is done , and then removal of older node groups later.

This command below shows an example upgrade path from the eksctl docs which makes use of the "--release-version" flag: 
eksctl upgrade nodegroup --name=managed-ng-1 --cluster=managed-cluster --release-version=1.19.6-20210310

Request and suggestion: 
1. What would it take to implement our AMIs similar to eks optimized AMIs, to be able to handle upgrades using the --release-version flag ? 

2. Will it be possible to tag our AMIs with a --release-version version and use same ? 

3.  Is there any known solution for upgrading node groups that do not use an optimized release-version ?

Please let me know if you have questions or need additional information. I am also available to chat if necessary. Thanks.

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

Closes #5006

Add a note explaining the workflow required for upgrading managed nodegroups that are deployed using custom AMIs.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: userdocs/src/usage/eks-managed-nodes.md [hunk 1]
```diff
@@ -183,6 +183,23 @@ To upgrade to a specific AMI release version instead of the latest version, pass
 eksctl upgrade nodegroup --name=managed-ng-1 --cluster=managed-cluster --release-version=1.19.6-20210310
 ```
 
+!!!note 
+    If the managed nodes are deployed using custom AMIs, the following workflow must me followed in order to deploy to a new version of the custom AMI.
```
Comment (cPu1) [inline userdocs/src/usage/eks-managed-nodes.md:187]:
```suggestion
    If the managed nodes are deployed using custom AMIs, the following workflow must be followed in order to deploy a new version of the custom AMI.
```

File: userdocs/src/usage/eks-managed-nodes.md [hunk 2]
```diff
@@ -183,6 +183,23 @@ To upgrade to a specific AMI release version instead of the latest version, pass
...
+    - initial deployment of the nodegroup must be done using a launch template. e.g.
+      ```yaml
+      managedNodeGroups:
+        - name: launch-template-ng
+        launchTemplate:
+        id: lt-1234
+        version: "2" #optional (uses the default version of the launch template if unspecified)
```
Comment (cPu1) [inline userdocs/src/usage/eks-managed-nodes.md:194]: This isn't indented properly.

**Review by cPu1 — CHANGES_REQUESTED** [review]

File: userdocs/src/usage/eks-managed-nodes.md [hunk 3]
```diff
@@ -183,6 +183,23 @@ To upgrade to a specific AMI release version instead of the latest version, pass
...
 
+!!!note 
+    If the managed nodes are deployed using custom AMIs, the following workflow must be followed in order to deploy a new version of the custom AMI.
+    - initial deployment of the nodegroup must be done using a launch template. e.g.
+      ```yaml
+      managedNodeGroups:
+        - name: launch-template-ng
+        launchTemplate:
+        id: lt-1234
+        version: "2" #optional (uses the default version of the launch template if unspecified)
```
Comment (TiberiuGC) [inline userdocs/src/usage/eks-managed-nodes.md:194]:
```suggestion
        - name: launch-template-ng
          launchTemplate:
          id: lt-1234
          version: "2" #optional (uses the default version of the launch template if unspecified)
```

File: userdocs/src/usage/eks-managed-nodes.md (hunk 2 above)
Comment (TiberiuGC) [reply userdocs/src/usage/eks-managed-nodes.md]: 8b1e6eb59bf015fabb4c308fce4a8a169f4e51ca

File: userdocs/src/usage/eks-managed-nodes.md [hunk 4]
```diff
@@ -183,6 +183,23 @@ To upgrade to a specific AMI release version instead of the latest version, pass
...
 
+!!!note 
+    If the managed nodes are deployed using custom AMIs, the following workflow must be followed in order to deploy a new version of the custom AMI.
+    - initial deployment of the nodegroup must be done using a launch template. e.g.
+      ```yaml
+      managedNodeGroups:
+        - name: launch-template-ng
+          launchTemplate:
+          id: lt-1234
+          version: "2" #optional (uses the default version of the launch template if unspecified)
```
Comment (TiberiuGC) [inline userdocs/src/usage/eks-managed-nodes.md:194]:
```suggestion
          launchTemplate:
            id: lt-1234
            version: "2" #optional (uses the default version of the launch template if unspecified)
```

File: userdocs/src/usage/eks-managed-nodes.md [hunk 5]
```diff
@@ -183,6 +183,23 @@ To upgrade to a specific AMI release version instead of the latest version, pass
...
+      ```yaml
+      managedNodeGroups:
+        - name: launch-template-ng
+          launchTemplate:
+            id: lt-1234
+            version: "2" #optional (uses the default version of the launch template if unspecified)
+      ```
+    - create a new version of the custom AMI (using AWS EKS console).
+    - create a new launch template version with the new AMI ID (using AWS EKS console).
+    - upgrade the nodes to the new version of the launch template. e.g.
```
[the remark this answers is not in the document]
Comment (TiberiuGC) [reply userdocs/src/usage/eks-managed-nodes.md:199]: 0cda2d07060ab189d7f38b7e8968f4de3e210fcd

**Review by Himangini — APPROVED** [review]
👍🏻

**Review by cPu1 — APPROVED** [review]

**Pull request merged**